src/app/shared/utils/health-status.ts

```diff
@@ -71,7 +71,11 @@ export function getNodeHealthStatus(n: Node): HealthStatus {
 export function getMachineDeploymentHealthStatus(md: MachineDeployment): HealthStatus {
   if (md.deletionTimestamp) {
     return new HealthStatus('Deleting', StatusIcon.Error);
-  } else if (md.status && md.status.availableReplicas === md.spec.replicas) {
+  } else if (
+    md.status &&
+    md.status.availableReplicas === md.spec.replicas &&
+    md.status.availableReplicas === md.status.updatedReplicas
```

Comment:
Just to confirm - both checks need to be valid to consider the MD as running?

Reply:
IMHO, yes. It's a valid scenario to have:
- spec.replicas=2
- status.availableReplicas=2
- status.updatedReplicas=0
Right after applying a MachineDeployment change, for example. But that should be "updating", not "running". You can try it out yourself on dev by updating a MachineDeployment, the status icon should stay green for some time even though the new nodes are already provisioning in the background.